backends/cuda-gen/ceed-cuda-gen-operator-build.cpp

```diff
@@ -1264,14 +1271,35 @@ extern "C" int CeedCudaGenOperatorBuild(CeedOperator op) {
     case CEED_EVAL_CURL:
       break; // TODO: Not implemented
     }
+    // Restriction
+    ierr = CeedElemRestrictionGetData(Erestrict, (void **)&restr_data); CeedChk(ierr);
+    data->indices.out[i] = restr_data->d_ind;
+    if (data->indices.out[i]) {
+      ierr = CeedElemRestrictionGetIMode(Erestrict, &imode); CeedChk(ierr);
+      code << "  // InterlaceMode: "<<CeedInterlaceModes[imode]<<"\n";
+      code << "  writeDofs"<<(imode==CEED_NONINTERLACED?"":"Transpose")<<dim<<"d<ncomp_out_"<<i<<",P_out_"<<i<<">(data, nnodes_out_"<<i<<", elem, indices.out["<<i<<"], r_v"<<i<<", d_v"<<i<<");\n";
+    } else {
+      bool backendstrides;
+      ierr = CeedElemRestrictionGetBackendStridesStatus(Erestrict,
+                                                        &backendstrides);
+      CeedChk(ierr);
+      CeedInt strides[3] = {1, elemsize, elemsize*ncomp};
```

Comment:
I have doubt about the performance of this data layout on GPU, I'll check this during benchmarking.